packages/expo-cellular/ios/EXCellular/EXCellularModule.m

```diff
@@ -76,14 +76,17 @@ - (CTCarrier *)carrier
   CTTelephonyNetworkInfo *netinfo = [[CTTelephonyNetworkInfo alloc] init];
   NSDictionary<NSString *, CTCarrier *> *serviceSubscriberCellularProviders;
   CTCarrier *carrier;
-  
+
   if (@available(iOS 12.0, *)) {
+    NSString *serviceCurrentRadioAccessTechnology;
+    serviceCurrentRadioAccessTechnology = netinfo.serviceCurrentRadioAccessTechnology.allKeys.firstObject;
```

Comment:
I'm not sure this will be more reliable. We're not guaranteed anything about the order of `serviceCurrentRadioAccessTechnology`'s keys, and the docs for this property say: 

> The entry associated with a key is nil if the service is not registered on any network.

so perhaps the corresponding entry in `serviceSubscriberCellularProviders` will be nil anyway.

Probably a more reliable -- but still non-deterministic (but good enough, I suspect) -- fix is to loop through `serviceSubscriberCellularProviders` and take the first non-nil value. And if they are all nil. there's no `CTCarrier` to return anyway.

Reply:
On my phone, `serviceSubscriberCelluarProviders` has 2 entries one of which is nil and `serviceCurrentRadioAccessTechnology` had 1 entry which corresponded with the non-nil one. I might be jumping to conclusions, but I took this to mean that there would be some reliability here.

Happy to rewrite it to be a loop if you think that's the better method.

Reply:
I'm worried about the part in the docs that says services can be not registered on any network. I don't know how this API works and all the complexities with different types of radios and multiple SIMs. 

Is it possible for a key in `serviceCurrentRadioAccessTechnology` to point to a nil value while pointing to a non-nil CTCarrier in `serviceSubscriberCellularProviders `? (It is possible for `netinfo.serviceCurrentRadioAccessTechnology` to be nil or empty, however.)

Reply:
I've rewritten this to use a loop and take the first CTCarrier in `serviceSubscriberCellularProviders` with a non-nil `carrierName`, and if non exists it falls back the "first" CTCarrier from the dictionary.